### PR TITLE
[FIX] web: Model base class can dispatch at willStart

### PR DESCRIPTION
**Pull request by kebeclibre**
Have a component that uses a model with the useModel hook
During the mount, make that model dispatch a command
i.e. in the willStart of the component

Before this commit, there was a timeout because the model waited for
the end of the rendering to yield, and the willStart waited for that
to yield in turn. That was a deadlock

After this commit, we trigger a rendering on the component
only when we are already mounted

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/tests/component_extension_tests.js [hunk 1]
```diff
@@ -248,5 +249,59 @@ odoo.define('web.component_extension_tests', function (require) {
...
+                }
+                async willStart() {
+                    await this.model.dispatch('loadState', 'tell me so');
+                    return super.willStart(...arguments);
+                }
+            }
+            Parent.env = makeTestEnvironment({}, () => Promise.resolve());
+            Parent.template = xml`<div t-esc="text" />`;
+
+            const fixture = document.body.querySelector('#qunit-fixture');
```
Comment (Arcasias) [inline addons/web/static/tests/component_extension_tests.js:288]: Maybe use `testUtils.prepareTarget()` ?

**Review by Arcasias — APPROVED** [review]

File: addons/web/static/tests/component_extension_tests.js [hunk 2]
```diff
@@ -285,20 +285,20 @@ odoo.define('web.component_extension_tests', function (require) {
...
-            await parent.mount(fixture);
+            await parent.mount(target);
             assert.strictEqual(
-                fixture.innerHTML,
+                target.innerHTML,
                 '<div>tell me so</div>'
             );
 
             await model.dispatch('loadState', 'set my soul on fire');
+            await testUtils.nextTick();
```
Comment (aab-odoo) [inline addons/web/static/tests/component_extension_tests.js:299]: to check

**Review by aab-odoo — COMMENTED** [review]
This is indeed what I had in mind. Does it work? We'll need to validate this with @ged-odoo anyway.

**Comment by C3POdoo** [comment]
Dear @kebeclibre,

Thank you for your contribution but the version saas-13.3 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**